Expand docs about adding hello-world to a new track (#388)
exercism/docs: building/tracks/new/add-first-exercise.md

- The stub file contains an almost-working implementation, but instead of "Hello, World!" it uses "Goodbye, Mars!"
- It has no `prerequisites`
- It has no `practices`

### Updating the config

Start by adding an entry to the `exercises.practice` array in the top-level `config.json` file.

```
{
  "exercises": {
    "practice": [
      {
        "uuid": "",
        "slug": "hello-world",
        "name": "Hello World",
        "practices": [],
        "prerequisites": [],
        "difficulty": 1
      }
    ]
  }
}
```

You can use the [Configlet][configlet] tool to get a UUID.
Download Configlet by running `bin/fetch-configlet` from the root of the repository.
Then generate a UUID using the `bin/configlet uuid` command.

### Generating required files

To implement the "Hello, World!" exercise, you'll need to create several files.
Which files exactly is described in the [Practice Exercises documentation](/docs/building/tracks/practice-exercises).

Most of the files can be added automatically by running Configlet's `sync` command:

```

> bin/configlet sync --update --docs --metadata --tests

Reviewer: ```suggestion
bin/configlet sync --update --yes --docs --metadata --exercise hello-world
bin/configlet sync --update --tests include --exercise hello-world
```

Unfortunately, you can't yet do this in a single command (looks like a bug).

Author: Ah, I was wondering about that, but hadn't gotten around to asking why it was this way.

```

> In addition to the generated files, you will to create a test suite, a stub solution that serves as the starting point for the student, and a sample solution that CI uses to ensure that the test suite is in a good state.

Reviewer: I feel like the bigger reason for having a sample solution is to prove that the exercise _can_ be solved. Maybe something like:

```suggestion
In addition to the generated files, you will to create a test suite, a stub solution that serves as the starting point for the student, and a sample solution that passes all the tests to verify it is possible to solve the exercise (CI will verify this).
```

> In order to create these files you need to make some decisions about filenames and file paths.

Reviewer: ```suggestion
In order to create these files, you need to make some decisions about filenames and file paths.
```

The rule of thumb is to use names that are idiomatic for the language, and where there are no strong preferences prefer shallower directory structures.
The sample solution will need to be identifiable by the CI script, so it's advisable to choose a generic basename that all exercises can use, e.g. `example`, `sample`, or `reference-solution`.

> 

Reviewer: I wonder if we should also steer people towards populating the root `config.json`'s `files` property, which can be used as a template for the syncing of the filepaths later on. But maybe that should be a separate item and be part of the next couple of exercises (makes more sense there probably). We could then also point people towards using the same `configlet sync` commands, but those would then also populate the files.

Author: I'm really torn. Getting configlet to sync those files is really handy, but I think coming up with the file names in the context of implementing a concrete exercise is a bit better than coming up with the paths up front and getting configlet to populate the files.

> But maybe that should be a separate item and be part of the next couple of exercises (makes more sense there probably).

I think that's my conclusion as well.

### Configuring the exercise

One you've decided on the filenames and paths, edit the `exercises/practice/hello-world/.meta/config.json` file to reflect those choices.

> Also add your GitHub username to the "authors" array.

Reviewer: ```suggestion
Also add your GitHub username to the `"authors"` array.
```
